assets/section-main-product.css

```diff
@@ -1366,14 +1366,79 @@ a.product__text {
 /* Product-thumbnail snippet */
 
 .product-media-container {
+  --aspect-ratio: var(--preview-ratio);
+  --ratio-percent: calc(1 / var(--aspect-ratio) * 100%);
+  position: relative;
   width: 100%;
   max-width: calc(100% - calc(var(--media-border-width) * 2));
 }
 
+.product-media-container.constrain-height {
+  /* amount to subtract from viewport height when calculating media height */
+  --viewport-offset: 400px;
+  /* smallest height the image is allowed to scale down to */
+  --constrained-min-height: 300px;
+  /* the larger of either the calculated viewport height or the fixed minimum */
+  --constrained-height: max(var(--constrained-min-height), calc(100vh - var(--viewport-offset)));
```

Comment:
I was leaning that way because there's not too much downside to keeping them around, but fully assumed there was a chance I'd get pushback in review. I tried to name the vars clearly but I also assumed I was too close to it to adequately judge if everything would be easy to decipher for others (or future me). I can remove them if we think the vars are clear enough as is. @ludoboludo @eugenekasimov any opinions?

Reply:
I ended up removing _most_. I left a couple in elsewhere that I felt warranted special mention